### PR TITLE
Capabilities Filtering

### PR DESCRIPTION
**Pull request by psieg**
#2 #3 #7

### CODE REVIEW

**Comment by LRueckert** [comment]
looks good to me

File: babelsberg/constraintinterpreter.js [hunk 1]
```diff
@@ -262,38 +264,38 @@ Object.subclass('Babelsberg', {
...
+                    constraints[i] = null;
+                } finally {
+                    Constraint.current = null;
+                }
+            };
+
+            var min = Number.MAX_VALUE;
+            var minIndex = -1;
+            for (var i = 0; i < constraints.length; i++) {
+                if (constraints[i] && constraints[i].oComparisonMetrics.time < min) {
```
Comment (j4yk) [inline babelsberg/constraintinterpreter.js:285]: Didn't you want to remove the Hungarian notation? ;-)

Comment (LRueckert) [inline babelsberg/constraintinterpreter.js:285]: i took care of that one in 3ac19b7b7f03e6ed04a600f44319926e8288a8a7

**Comment by psieg** [comment]
yaaay, green tests! Comment or merge ;)

File: babelsberg/constraintinterpreter.js [hunk 2]
```diff
@@ -1032,31 +1070,46 @@ Object.subclass('ConstrainedVariable', {
...
+
+                if (s.weight > solver.weight) {
+                    solver = s;
+                }
+            }
+        }.bind(this));
+
+        if (solver.fake && solvers.length == 1) {
+            // if there is only one disabled solver, use that one even if disabled
+            // why? to make tests green, that's why (because nobody cared about
```
Comment (j4yk) [inline babelsberg/constraintinterpreter.js]: Should we have  this rant removed before the merge?

Comment (LRueckert) [inline babelsberg/constraintinterpreter.js]: i guess so

File: babelsberg/constraintinterpreter.js [hunk 3]
```diff
@@ -497,29 +531,29 @@ Object.subclass('Constraint', {
...
             }
             this._enabled = true;
-            var nBegin = performance.now();
+            this.constraintvariables.each(function(v) {v._resetIsSolveable();});
+            console.time('solve');
             this.solver.solve();
-            var nEnd = performance.now();
-            console.log("Time to Solve in enable with solver below:" + (nEnd - nBegin) + " ms");
-            console.log(this.solver);
+            var time = console.timeEnd('solve');
```
Comment (j4yk) [inline babelsberg/constraintinterpreter.js]: console.timeEnd gibt keinen Wert zurück (undefined), daher steht auf der Konsole jetzt sowas wie "Time to solve in enable with solver Cassowary: undefined ms"

File: babelsberg/constraintinterpreter.js (hunk 2 above)
Comment (psieg) [inline babelsberg/constraintinterpreter.js]: before removing this we should properly consider whether this should be in a final version or not. It seems wrong, the rant wants to indicate the tests might rely on incorrect behavior ;)

File: babelsberg/constraintinterpreter.js (hunk 3 above)
Comment (psieg) [inline babelsberg/constraintinterpreter.js]: ~~wann gibt `console.timeEnd` keinen Wert zurück?~~
Looks like we need yet another timing mechanism....
According to chrome docs  `console.timeEnd` prints the time on console (with name) but does _not_ return it

**Comment by psieg** [comment]
I guess we're going to go back to `performance.now` and use `DateTime` if not available: http://www.sitepoint.com/discovering-the-high-resolution-time-api/

**Comment by gersseba** [comment]
:+1:

**Pull request merged**